packages/toolkit/src/query/react/buildHooks.ts

```diff
@@ -451,11 +457,26 @@ type GenericPrefetchThunk = (
  */
 export function buildHooks<Definitions extends EndpointDefinitions>({
   api,
-  moduleOptions: { batch, useDispatch, useSelector, useStore },
+  moduleOptions: {
+    batch,
+    useDispatch,
+    useSelector,
+    useStore,
+    unstable__sideEffectsInRender,
+    unstable__suspense,
+  },
 }: {
   api: Api<any, Definitions, any, any, CoreModule>
   moduleOptions: Required<ReactHooksModuleOptions>
 }) {
+  const usePossiblyImmediateEffect: (
+    effect: () => void | undefined,
+    deps?: DependencyList
+  ) => void =
+    unstable__sideEffectsInRender || unstable__suspense
```

Comment:
Is the separate `unstable__sideEffectsInRender` needed here, maybe it could just be "if you want to fetch inside of components on the server, you use Suspense"?

Reply:
I wanted to enable a "fetch in render, but do not throw" use case here, so I'd prefer to keep them separated.

Reply:
Ah, I get it, I missed that this wasn't throwing! As I mentioned in the issue I'm wary of that use case personally and don't see any benefit over using the Suspense paradigm, but that's up to you of course and this _is_ clearly marked as unstable. 😄